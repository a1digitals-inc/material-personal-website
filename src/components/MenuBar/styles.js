import { makeStyles } from "@material-ui/core/styles";

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    marginLeft: drawerWidth,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  hide: {
    display: "none",
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    background: theme.palette.background.default,
    width: drawerWidth,
  },
  drawerHeader: {
    display: "flex",
    alignItems: "center",
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: "flex-end",
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
    marginLeft: -drawerWidth,
  },
  contentShift: {
    marginLeft: 0,
  },
}));

export default useStyles;
