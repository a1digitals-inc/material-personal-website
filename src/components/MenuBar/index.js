import React, { useState } from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { useLocation } from "react-router-dom";
import { useTheme } from "@material-ui/core/styles";
import Drawer from "@material-ui/core/Drawer";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import List from "@material-ui/core/List";
import Divider from "@material-ui/core/Divider";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import ChevronRightIcon from "@material-ui/icons/ChevronRight";
import Container from "@material-ui/core/Container";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import { menuItems } from "../../config/dataApi";
import useStyles from "./styles";
import ListItemLink from "../ListItemLink";

const MenuBar = ({ children }) => {
  const classes = useStyles();
  const theme = useTheme();

  let location = useLocation();

  const [open, setOpen] = useState(false);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <div className={classes.root}>
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open,
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            edge="start"
            className={clsx(classes.menuButton, open && classes.hide)}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="overline" display="block">
            {location.pathname.replace(/[^a-z0-9]/gi, "")}
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer
        className={classes.drawer}
        variant="persistent"
        anchor="left"
        open={open}
        classes={{
          paper: classes.drawerPaper,
        }}
      >
        <div className={classes.drawerHeader}>
          <IconButton onClick={handleDrawerClose}>
            {theme.direction === "ltr" ? (
              <ChevronLeftIcon />
            ) : (
              <ChevronRightIcon />
            )}
          </IconButton>
        </div>
        <Divider />
        <List>
          {menuItems.map((menuItem) => (
            <ListItemLink
              key={menuItem.to}
              to={menuItem.to}
              primary={menuItem.primary}
              icon={<FontAwesomeIcon icon={menuItem.icon} />}
              handleClick={handleDrawerClose}
            />
          ))}
        </List>
      </Drawer>
      <main
        className={clsx(classes.content, {
          [classes.contentShift]: open,
        })}
      >
        <div className={classes.drawerHeader} />
        <Container component="main">{children}</Container>
      </main>
    </div>
  );
};

MenuBar.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MenuBar;
