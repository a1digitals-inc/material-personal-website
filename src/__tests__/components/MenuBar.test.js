import React from "react";
import { shallow } from "enzyme";

import MenuBar from "../../components/MenuBar";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/",
  }),
}));

it("renders without crashing", () => {
  const wrapper = shallow(
    <MenuBar>
      <div className="unique" />
    </MenuBar>
  );
  expect(wrapper).toMatchSnapshot();
});
