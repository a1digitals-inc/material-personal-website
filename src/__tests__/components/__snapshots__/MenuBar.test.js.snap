// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    className=""
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="open drawer"
        className="makeStyles-menuButton-3"
        color="inherit"
        edge="start"
        id="open-menubar-button"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    className="makeStyles-drawer-5"
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-6",
      }
    }
    id="menubar-drawer"
    open={false}
    variant="persistent"
  >
    <div
      className="makeStyles-drawerHeader-7"
    >
      <WithStyles(ForwardRef(IconButton))
        id="close-menubar-button"
        onClick={[Function]}
      >
        <ChevronLeftIcon />
      </WithStyles(ForwardRef(IconButton))>
    </div>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(List))>
      <ListItemLink
        handleClick={[Function]}
        key="/"
        primary="Home"
        to="/"
      />
      <ListItemLink
        handleClick={[Function]}
        key="/portfolio"
        primary="Portfolio"
        to="/portfolio"
      />
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(Drawer))>
  <main
    className="makeStyles-content-8"
  >
    <div
      className="makeStyles-drawerHeader-7"
    />
    <WithStyles(ForwardRef(Container))
      component="main"
    >
      <div
        className="unique"
      />
    </WithStyles(ForwardRef(Container))>
  </main>
</div>
`;
